packages/props/src/filterProps.ts

```diff
@@ -0,0 +1,75 @@
+import { createMemo } from "solid-js";
+import { propTraps } from "./propTraps";
+
+/**
+ * An alternative primitive to Solid's [splitProps](https://www.solidjs.com/docs/latest/api#splitprops) allowing you to create a new props object with only the property names that match the predicate.
+ *
+ * The predicate is run for every property read lazily, instead of calculated eagerly like the original splitProps. Any signal accessed within the `predicate` will be tracked, and `predicate` re-executed if changed.
+ *
+ * @param props The props object to filter.
+ * @param predicate A function that returns `true` if the property should be included in the filtered object.
+ * @returns A new props object with only the properties that match the predicate.
+ *
+ * @example
+ * ```tsx
+ * const dataProps = filterProps(props, key => key.startsWith("data-"));
+ * <div {...dataProps} />
+ * ```
+ */
+export function filterProps<T>(props: T, predicate: (key: keyof T) => boolean): T {
+  return new Proxy(
+    {
+      get(property: string | number | symbol) {
+        return predicate(property as keyof T) ? props[property as keyof T] : undefined;
+      },
+      has(property: string | number | symbol) {
+        return predicate(property as keyof T) ? property in props : false;
```

Comment:
Maybe you don't want to needlessly run the predicate if the property is not even existing in the original props?